### PR TITLE
Support post_filter for query webapi

### PR DESCRIPTION
**Pull request by sengineer0**
Ref: https://github.com/biothings/biothings.api/issues/208

This PR tried to add support `post_filter` feature for the webapi's query, if provided, the search hits are filtered after the aggregations are calculated (https://www.elastic.co/guide/en/elasticsearch/reference/8.1/filter-search-results.html#post-filter)
That params should be provided in query_string format (https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html)

Example
```
GET /v1/query?q=cyclin dependent kinase 2&aggs=type_of_gene&post_filter=taxid:216574
```

- Response:
```
       {
            'facets': {
                'type_of_gene': {
                    '_type': 'terms',
                    'missing': 0,
                    'other': 0,
                    'terms': [{'count': 79, 'term': 'protein-coding'}],
                    'total': 79
                }
            },
            'hits': [...],
            'max_score': 0.4116186,
            'took': 7,
            'total': 1
        }
```

### CODE REVIEW

File: biothings/web/query/builder.py [hunk 1]
```diff
@@ -360,10 +357,19 @@ def apply_extras(self, search, options):
         # https://www.elastic.co/guide/en/elasticsearch/
         # reference/current/index-modules.html
 
+        # Feature: post_filter
+        # -- implementation using query string matching
+        # Ref: https://www.elastic.co/guide/en/elasticsearch/reference/8.1/filter-search-results.html#post-filter
+        if options.aggs and options.post_filter:
```
Comment (newgene) [inline biothings/web/query/builder.py]: @sengineer0 can `post_filter` work without `aggs`? Please verify it on ES.

If it works without `aggs`, will `post_filter` changes the `total` value in the response?

File: biothings/web/settings/default.py [hunk 2]
```diff
@@ -104,32 +104,34 @@
...
     'GET': {'id': {'type': str, 'path': 0, 'required': True}},
-    'POST': {'id': {'type': list, 'max': 1000, 'required': True, 'alias': 'ids'}}
+    'POST': {'id': {'type': list, 'max': 1000, 'required': True, 'alias': 'ids'}},
 }
 QUERY_KWARGS = {
     '*': COMMON_KWARGS.copy(),
     'GET': {
         'q': {'type': str, 'default': None},
         'aggs': {'type': list, 'max': 1000, 'alias': 'facets'},
+        'post_filter': {'type': list, 'max': 1000},
```
Comment (newgene) [inline biothings/web/settings/default.py]: I know the original implementation set `type` as list, but I feel it does not need to, a string should be good enough. What do you think?

File: biothings/web/query/builder.py (hunk 1 above)
Comment (sengineer0) [reply biothings/web/query/builder.py]: ok, let's me check it

File: biothings/web/settings/default.py (hunk 2 above)
Comment (sengineer0) [reply biothings/web/settings/default.py]: I think the `post_filter` can work with string type, but still need to double check it.

File: biothings/web/query/builder.py (hunk 1 above)
Comment (sengineer0) [reply biothings/web/query/builder.py]: I tried to remove `aggs`, and the `post_filter` still worked, and it changed `total` value.
Although the ES's documentation didn't tell that `post_filter` must be used with `aggs`, I am prefer to only use `post_filter` with `aggs`.

File: biothings/web/settings/default.py (hunk 2 above)
Comment (sengineer0) [reply biothings/web/settings/default.py]: Yes, we can change `post_filter` to str type, and it still works.

Comment (sengineer0) [reply biothings/web/settings/default.py]: @newgene

File: biothings/web/query/builder.py (hunk 1 above)
Comment (sengineer0) [reply biothings/web/query/builder.py]: @newgene

Comment (newgene) [reply biothings/web/query/builder.py]: ok, thanks for confirming. In this case, let's remove the `aggs` requirements. I agree it's typically used together with `aggs`, but I think we have a use case for `post_filter` without `aggs` as well.

File: biothings/web/settings/default.py (hunk 2 above)
Comment (newgene) [reply biothings/web/settings/default.py]: ok, let's change it to string then

File: biothings/web/query/builder.py (hunk 1 above)
Comment (sengineer0) [reply biothings/web/query/builder.py]: @newgene  ok. no problem with me. I will update my PR.

File: biothings/web/settings/default.py (hunk 2 above)
Comment (sengineer0) [reply biothings/web/settings/default.py]: It already updated. :)

File: biothings/web/query/builder.py (hunk 1 above)
Comment (sengineer0) [reply biothings/web/query/builder.py]: It's updated. @newgene

**Pull request merged**